runelite-client/src/main/java/net/runelite/client/RuneLite.java

```diff
@@ -146,6 +146,16 @@ public UpdateCheckMode convert(String v)
 			System.exit(0);
 		}
 
+		if (RuneLite.getOptions().has("developer-mode"))
+		{
+			boolean assertions = false;
+			assert assertions = true;
```

Comment:
Google says that https://docs.oracle.com/javase/6/docs/api/java/lang/Class.html#desiredAssertionStatus%28%29 is the correct way to test for assertions being enabled.